### PR TITLE
Allow queries over arbitrary net.Conn

### PR DESCRIPTION
**Pull request by tcolgate**
This introduces NewLivestatusFromDialer that takes a custom dial
function.

In our local setup we have wrapped livestatus in SSL and do some
psot connection checking of the cert name checking. The above patch
means I can wrap all that up and just do livestatus over the
resulting net.Conn

If you're interested in upstreaming this, I'll add some test etc,
I just wanted to see if you were interested in the idea

### CODE REVIEW

File: livestatus.go [hunk 1]
```diff
@@ -19,3 +22,10 @@ func NewLivestatus(network, address string) *Livestatus {
 		address: address,
 	}
 }
+
+// NewTLSLivestatus creates a new binding instance.
```
Comment (vbatoufflet) [inline livestatus.go]: The comment should start with `NewLivestatusWithDialer` and mention that it uses an external dialer.

File: query.go [hunk 2]
```diff
@@ -82,14 +82,20 @@ func (q *Query) Exec() (*Response, error) {
...
+		n, err := conn.Read(data)
 		if err == io.EOF {
 			break
 		} else if err != nil {
 			return nil, err
 		}
 
 		buf.Write(bytes.TrimRight(data, "\x00"))
+
+		// New lines signals end of conten. Helps if
```
Comment (vbatoufflet) [inline query.go]: The is a typo to the “content” word.

**Comment by vbatoufflet** [comment]
Hi @tcolgate,

Thanks for the PR. I just made some observations on its content just to fix some comments before merging it.

Regards.

**Comment by tcolgate** [comment]
Sorry about that, I was clearly in a bit of a rush! I've converted the comment into actual English.

**Comment by vbatoufflet** [comment]
Great, thanks. PR has been merged.

**Pull request merged**